speech-to-text/functions/src/index.ts

```diff
@@ -53,23 +55,42 @@ export const transcribeAudio = functions.storage
     logs.start();
     const {contentType} = object; // the MIME type
 
+    /** Return early if no file name is specified */
+    if (object.name === undefined) {
+      logs.undefinedObjectName(object);
+      return;
+    }
+
+    /** Set a sanitizd document ID */
+    const sanitizedDocumentId = object.name?.replace(/\//g, '_');
```

Comment:
Yeah I agree, the reason for the custom documentId was so that it could referenced easily from a client for example.

Could there a be a way for users to look up the transaction document based on the file name provided?

Reply:
I think they could just use a query and a where clause, like `where('fileName', '==', '<original storage file>') if that makes sense

Reply:
Agreed, updated!